host/xtest/regression_1000.c

```diff
@@ -3091,6 +3091,7 @@ static void xtest_tee_test_1037(ADBG_Case_t *c)
 ADBG_CASE_DEFINE(regression, 1037, xtest_tee_test_1037,
 		 "Remote attestation");
 
+#if defined(CFG_MEMTAG)
```

Comment:
Please do a runtime check instead, same as the attestation test (see line 3067) so that xtest doesn't depend on the OP-TEE configuration too much. Historically we have used ifdefs but there are use cases to avoid that (common root FS with the test suite, used for different platforms that may or may not enable the same optee_os options).

Reply:
Ack - still have to use ifdefs for the TA side though.

Reply:
Why?

Reply:
Sorry my reference to line 3067 was incorrect, in fact I don't understand what you are trying to accomplish. The code is already supposed to skip the MTE tests if MTE is not supported.

Reply:
I got compilation like this in `os_test.c`:

```
| os_test.c: In function 'ta_entry_memtag_use_after_free':
| os_test.c:1473:14: error: implicit declaration of function 'memtag_is_enabled' [-Werror=implicit-function-declaration]
|  1473 |         if (!memtag_is_enabled())
|       |              ^~~~~~~~~~~~~~~~~
| os_test.c:1473:14: warning: nested extern declaration of 'memtag_is_enabled' [-Wnested-externs]
| os_test.c: In function 'ta_entry_memtag_invalid_tag':
| os_test.c:1493:14: error: implicit declaration of function 'memtag_get_tag' [-Werror=implicit-function-declaration]
|  1493 |         if (!memtag_get_tag(p)) {
|       |              ^~~~~~~~~~~~~~
| os_test.c:1493:14: warning: nested extern declaration of 'memtag_get_tag' [-Wnested-externs]
| os_test.c:1499:14: error: implicit declaration of function 'memtag_insert_tag' [-Werror=implicit-function-declaration]
|  1499 |         p2 = memtag_insert_tag(p, (memtag_get_tag(p) + 1) & 0xf);
|       |              ^~~~~~~~~~~~~~~~~
| os_test.c:1499:14: warning: nested extern declaration of 'memtag_insert_tag' [-Wnested-externs]
| os_test.c:1499:12: warning: assignment to 'uint32_t *' {aka 'unsigned int *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
|  1499 |         p2 = memtag_insert_tag(p, (memtag_get_tag(p) + 1) & 0xf);
|       |            ^
```

Reply:
Okay, I think I got it - I was using `3.17.0` of `optee-os`, but `optee-test` was on latest so there is a mismatch.

Reply:
No worries, these things happen! :wink: